src/Diagnostics.RuntimeHost/Utilities/RequestBodyValidator.cs

```diff
@@ -20,5 +23,32 @@ public static bool ValidateRequestBody(JToken jsonBody, string[] fieldNames, out
             outputMessage = string.Concat(validationResults);
             return !(validationResults.Count() > 0);
         }
+
+        public static bool ValidateRequestBody(IRequestBodyBase body, out string outputMessage)
+        {
+            outputMessage = "";
+            if (body == null)
+            {
+                outputMessage = "Missing request body";
+                return false;
+            }
+
+            List<string> missingProperties = new List<string>();
+
+            foreach (var property in body.GetType().GetProperties())
+            {
+                if (property.GetValue(body) == null)
+                {
+                    missingProperties.Add(property.Name);
+                }
+            }
+
+
+            var message = "The following fields are missing from the request body: ";
+            foreach (var x in missingProperties) message += x + ", ";
+            outputMessage = message.Remove(message.Length - 2);
+
+            return !(missingProperties.Count > 0);
```

Comment:
Two points:
1. You can check if Count > 0 and return true before crafting outputMessage. This way you can save cost when validated successfully.
2. For line 47 - 49, you could write this way: outputMessage = message + String.Join(", ", missingProperties);
String.Join() is quite useful because it inserts delimiters between elements. You don't need Remove(message.Length - 2).